### PR TITLE
BAU: fixing tests

### PR DESCRIPTION
**Pull request by ed0906**

### CODE REVIEW

File: app/controllers/LegalChallengeDetailsController.scala [hunk 1]
```diff
@@ -46,31 +46,29 @@ class LegalChallengeDetailsController @Inject()(
...
-  def onPageLoad(mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData) {
-    implicit request =>
+  def onPageLoad(mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData) { implicit request =>
 
-      val preparedForm = request.userAnswers.get(LegalChallengeDetailsPage) match {
-        case None => form
-        case Some(value) => form.fill(value)
-      }
+    val preparedForm = request.userAnswers.get(LegalChallengeDetailsPage) match {
+      case None => form
```
Comment (ed0906) [inline app/controllers/LegalChallengeDetailsController.scala]: Didnt we plan to change this to a `_`
I'm happy either way

File: app/controllers/RegisteredAddressForEoriController.scala [hunk 2]
```diff
@@ -39,36 +39,35 @@ class RegisteredAddressForEoriController @Inject()(appConfig: FrontendAppConfig,
...
 
-  def onSubmit(mode: Mode): Action[AnyContent] = (identify andThen getData).async {
-    implicit request =>
+  def onSubmit(mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData).async { implicit request =>
 
-      form.bindFromRequest().fold(
-        (formWithErrors: Form[_]) =>
-          Future.successful(BadRequest(registeredAddressForEori(appConfig, formWithErrors, mode))),
-        value => {
-          val updatedAnswers = request.userAnswers.getOrElse(UserAnswers(request.internalId)).set(RegisteredAddressForEoriPage, value)
```
Comment (ed0906) [inline app/controllers/RegisteredAddressForEoriController.scala]: How come this was removed?

File: test/forms/ConfidentialInformationFormProviderSpec.scala [hunk 3]
```diff
@@ -27,20 +28,11 @@ class ConfidentialInformationFormProviderSpec extends StringFieldBehaviours {
...
       form,
       fieldName,
-      stringsWithMaxLength(maxLength)
-    )
-
-    behave like fieldWithMaxLength(
-      form,
-      fieldName,
-      maxLength = maxLength,
-      lengthError = FormError(fieldName, lengthKey, Seq(maxLength))
```
Comment (ed0906) [inline test/forms/ConfidentialInformationFormProviderSpec.scala]: Did we mean to delete these? They look like valid assertions to me

File: test/forms/DescribeYourItemFormProviderSpec.scala [hunk 4]
```diff
@@ -54,20 +55,11 @@ class DescribeYourItemFormProviderSpec extends StringFieldBehaviours {
...
       form,
       fieldName,
-      stringsWithMaxLength(maxLength)
-    )
-
-    behave like fieldWithMaxLength(
-      form,
-      fieldName,
-      maxLength = maxLength,
-      lengthError = FormError(fieldName, lengthKey, Seq(maxLength))
```
Comment (ed0906) [inline test/forms/DescribeYourItemFormProviderSpec.scala]: Did we mean to delete these?

File: app/controllers/RegisteredAddressForEoriController.scala (hunk 2 above)
Comment (ed0906) [reply app/controllers/RegisteredAddressForEoriController.scala]: It looks to me like it's protecting against `request.userAnswers` being `None` which I imagine will happen on the first time you go through the journey. Need to double check this still works

File: test/forms/ConfidentialInformationFormProviderSpec.scala (hunk 3 above)
Comment (gokyo) [reply test/forms/ConfidentialInformationFormProviderSpec.scala]: `ConfidentialInformationFormProvider` does not have any requirement on the length 
So I removed the scenario on the text length

File: test/forms/DescribeYourItemFormProviderSpec.scala (hunk 4 above)
Comment (gokyo) [reply test/forms/DescribeYourItemFormProviderSpec.scala]: `DescribeYourItemFormProvider . field2` does not have any requirement on the length 
So I removed the scenario on the text length

File: app/controllers/LegalChallengeDetailsController.scala (hunk 1 above)
Comment (ed0906) [reply app/controllers/LegalChallengeDetailsController.scala]: Same :)

File: test/views/CommodityCodeDigitsViewSpec.scala [hunk 5]
```diff
@@ -38,11 +38,46 @@ class CommodityCodeDigitsViewSpec extends StringViewBehaviours {
...
-      createViewUsingForm,
-      messageKeyPrefix,
-      routes.CommodityCodeDigitsController.onSubmit(NormalMode).url
-    )
+    "behave like a page with a string value field" when {
+      "rendered" must {
+
+        "contain a label for the value" in {
+          val doc = asDocument(createViewUsingForm(form))
+          assertContainsLabel(doc, "value", messages(s"$messageKeyPrefix.label"))
```
[the remark this answers is not in the document]
Comment (pllaurado) [reply test/views/CommodityCodeDigitsViewSpec.scala]: you mean, I should reuse something which is asserting on a hidden labels...  This is why is not working.... LongShort story, Every single field will have a label hidden with the value from the heading unless it have explicitly a label... 
Then doing logic to check if there is label defined on our messages or not...

File: app/controllers/RegisteredAddressForEoriController.scala (hunk 2 above)
Comment (ed0906) [reply app/controllers/RegisteredAddressForEoriController.scala]: Just give it a go in the UI, if it works then happy days

File: test/forms/ConfidentialInformationFormProviderSpec.scala (hunk 3 above)
Comment (ed0906) [reply test/forms/ConfidentialInformationFormProviderSpec.scala]: Sure

File: test/forms/DescribeYourItemFormProviderSpec.scala (hunk 4 above)
Comment (ed0906) [reply test/forms/DescribeYourItemFormProviderSpec.scala]: Sure

File: test/views/CommodityCodeDigitsViewSpec.scala (hunk 5 above)
[the remark this answers is not in the document]
Comment (ed0906) [reply test/views/CommodityCodeDigitsViewSpec.scala]: tut tut

**Pull request merged**